tests/tests.js

```diff
@@ -395,6 +395,31 @@ describe('common usage', function() {
 			'Buffer.from([\n\t19,\n\t55,\n\t66\n])',
 			'Stringifying a Buffer with `compact: false`'
 		);
+		// Date
+		assert.equal(
+			jsesc(
+				new Date('2020-04-03T15:34:00Z')
+			),
+			'new Date(1585928040000)'
+		);
+		assert.equal(
+			jsesc(
+				new Date('2020-04-03T15:34:00Z'),
+				{
+					'numbers': 'hexadecimal'
+				}
+			),
+			'new Date(0x17140AD6E40)'
+		);
+		assert.equal(
+			jsesc(
+				new Date('2020-04-03T15:34:00Z'),
+				{
+					'json': true
+				}
+			),
+			'"2020-04-03T15:34:00.000Z"'
```

Comment:
This depends on how we want to deal with https://github.com/mathiasbynens/jsesc/issues/57, Is this what you'd expect when passing a Date/Map/Set while using `json: true`? Or would you want to throw an exception instead?

Reply:
Either way this would need to be documented in the README under the `json` option.

Reply:
This is the behavior of `toJSON`. I'd like to add an additional flag in a subsequent PR (or, I suppose, in this PR) to let the user override that behavior independent of the JSON mode.

Reply:
Maybe we could include an `interpret` option that accepts values like `expect-json`, `coerce-json`, and `javascript` to switch between the options. `expect-json` would throw on non-JSON-supported values; `coerce-json` would invoke `toJSON` and delegate to `JSON.stringify`'s behavior in other cases; `javascript` would do its best to serialize all (non-circular) values for use in a js context.

I don't need `expect-json` for my own work, but `coerce-json` would be extremely useful. I've also been wondering if we could introduce a `replacer`-type option for both transforming and extending the supported values - e.g. we need strict `coerce-json` semantics, but also want to support the `undefined` value. If you're amenable to that, I'll see what that would take.

Reply:
The way I've used `jsesc` is always very controlled, in that: if there's a `Date` object, a `Map`, a `Set`, or a `BigInt` somewhere in the input, and I am requesting JSON output via `json: true`, that's probably a mistake. I've only used `jsesc` in cases where I want `JSON.parse(output)` to produce an equivalent object again.

So I defer to you on this. If you have a use case for it, that's great!